nnoir/nnoir/functions/linear.py

```diff
@@ -10,7 +10,7 @@ def __init__(self, inputs, outputs, **params):
         super(Linear, self).__init__(inputs, outputs, params, required_params, optional_params)
 
     def run(self, x):
-        result = x.reshape(x.shape[0], -1).dot(np.transpose(self.params['W']))
+        result = np.matmul(x, self.params["W"].T)
```

Comment:
I have to improve the method for test.

Reply:
A nnoir's Linear respects chainer's one.
```
>>> x = np.ones((2,3,4), dtype=np.float32)
>>> W = np.ones((5,12), dtype=np.float32)
>>> chainer.functions.linear(x,W).data.shape
(2, 5)
```
This behavior is different from `np.matmul`.

Reply:
thx. I have to modify it.

Reply:
I revert it in fb3ed48